static/js/click_handlers.js

```diff
@@ -83,6 +83,24 @@ $(function () {
 
     // MESSAGE EDITING
 
+    $('body').on('focus', '.message_edit_topic', function (e) {
+        var row = current_msg_list.get_row(rows.id($(this).closest(".message_row")));
+        var message = current_msg_list.get(rows.id(row));
+        var original_topic = message.subject;
+        if (original_topic === "(no topic)") {
+            $(this).val('');
+        }
+    });
+
+    $('body').on('click', '.message_edit_topic', function (e) {
+        var row = current_msg_list.get_row(rows.id($(this).closest(".message_row")));
+        var message = current_msg_list.get(rows.id(row));
+        var original_topic = message.subject;
+        if (original_topic === "(no topic)") {
+            $(this).val('');
+        }
+    });
+
```

Comment:
I think these click/focus handlers shouldn't be needed anymore. The "if original_topic is compose.empty_topic_placeholder(), change the value" logic is still needed, but that can go directly in message_edit.edit_message, where we have easier access to the `row`.